lib/browser/event-target.ts

```diff
@@ -31,6 +33,326 @@ export function eventTargetPatch(_global: any) {
 
   for (let i = 0; i < apis.length; i++) {
     const type = _global[apis[i]];
-    patchEventTargetMethods(type && type.prototype);
+    patchEventTargetMethodsOptimized(type && type.prototype);
+  }
+}
+
+const TRUE_STR = 'true';
+const FALSE_STR = 'false';
+// an identifier to tell ZoneTask do not create a new invoke closure
+const OPTIMIZED_ZONE_EVENT_TASK = zoneSymbol('optimizedZoneEventTask');
+
+const zoneSymbolEventNames: any = {};
+const globalSources: any = {};
+
+const _global: any =
```

Comment:
Could we get this from the patch function global? It would be better if we checked for global in exactly one place.